addons/survey/controllers/main.py

```diff
@@ -150,6 +150,8 @@ def _redirect_with_error(self, access_data, error_key):
             return request.render("survey.survey_auth_required", {'survey': survey_sudo, 'redirect_url': redirect_url})
         elif error_key == 'answer_deadline' and answer_sudo.access_token:
             return request.render("survey.survey_closed_expired", {'survey': survey_sudo})
+        elif error_key in ['answer_wrong_user', 'token_wrong']:
+            return request.render("survey.survey_access_error", {'survey': survey_sudo})
 
         return request.redirect("/")
```

Comment:
Hello @thiryr 

Actually Yes, it prevents and also informs the user about it.
and it's up to the user if they want to go back or not.

Thanks

Reply:
But if we were redirected to '/' anyway they shouldn't have been able to interact with the survey at all right?

As far as I know the issue is that sometimes people join surveys but some/all the answers are already filled-out (as if they shared an answer token, which should not be possible), maybe we're thinking of different issues?

Reply:
Hello @thiryr 

Yes, the User will be redirected to  '/'.
But, it will confuse the user and they won't be able to know what happened.
so that we have created a template with an appropriate warning message.

Thanks

Reply:
As discussed over discord there was a mixup in the tasks

This is related to https://github.com/odoo/odoo/pull/97895 as you pointed out.

I'll go on with the validation going from there :)